### PR TITLE
Added color variables to Header

### LINKED ISSUE #366: Header needs theme variables for color attributes
# Feature Request
In the current Header implementation, we cannot change the colors for a theme. We need this for the lowlight theme, as just using default colors makes the text unreadable.

## Description
Add variables to theme header's and subheader's background and any other colors that might exist.

### PR DESCRIPTION
**Pull request by cwalten**
### Summary
Color variables have been added to the Header component. Resolves #366 

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-clinical-header/src/Header.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
 :local {
   .flex-header {
-    background-color: #f4f4f4;
-    border-bottom-color: #dedfe0;
+    background-color: var(--terra-header-background-color, #f4f4f4);
```
Comment (emilyrohrbough) [inline packages/terra-clinical-header/src/Header.scss]: Can we updates these to be name-spaced to terra-clinical?

Comment (cwalten) [reply packages/terra-clinical-header/src/Header.scss]: Based on other scss files in this repo, I omitted clinical from the name. I have no problem adding the name spacing here, just wanted to make sure we are aware we are breaking that pattern.

Comment (emilyrohrbough) [reply packages/terra-clinical-header/src/Header.scss]: oh no... IMO  that pattern that should have been followed for the clinical components.
@ryanthemanuel @mjhenkes thoughts?

Comment (ryanthemanuel) [reply packages/terra-clinical-header/src/Header.scss]: Yes, let's go ahead make this one terra-clinical.  The other ones probably should be as well, but that would be non-passive at this point.

**Pull request merged**